lua/fpp/server/ownability.lua

```diff
@@ -267,51 +267,58 @@ end
 Networking
 ---------------------------------------------------------------------------]]
 util.AddNetworkString("FPP_TouchabilityData")
--- Sends 32 + 32 + 5 + 20 = 89 bits of ownership data per entity
+-- Sends 13 + 8 + 5 + 20 = 46 bits of ownership data per entity
 local function netWriteEntData(ply, ent)
-    -- EntIndex for when it's out of the PVS of the player
-    net.WriteUInt(ent:EntIndex(), 32)
+    -- EntIndex for when it's out of the PVS of the player`
+    net.WriteUInt(ent:EntIndex(), 13)
 
     local owner = ent:CPPIGetOwner()
-    net.WriteUInt(IsValid(owner) and owner:EntIndex() or -1, 32)
+    net.WriteUInt(IsValid(owner) and owner:EntIndex() or -1, 8)
     net.WriteUInt(ent.FPPRestrictConstraint and ent.FPPRestrictConstraint[ply] or ent.FPPCanTouch[ply], 5) -- touchability information
     net.WriteUInt(ent.FPPConstraintReasons and ent.FPPConstraintReasons[ply] or ent.FPPCanTouchWhy[ply], 20) -- reasons
 end
 
-function FPP.plySendTouchData(ply, ents)
-    local count = #ents
-
-    if count == 0 then return end
+local touchabilityDataQueue = {}
+local function startNetWriteQueue()
```

Comment:
With timers I have a fear of funny race conditions. Think of e.g. this function returning because the timer exists, but right after, the other part of the code decides to remove the timer, even though the queue is not worked away yet.

The tricky thing about such bugs is that they're near impossible to reproduce, but somewhere, among the hundreds of servers running FPP, someone will hit it.

Perhaps a simpler solution can be made without creating and removing timers. 

How about the following idea:
Keep track of a per player count of "amount of entities sent this tick to this player". Also keep track of a per player queue. That queue is a table with number indices, and ent data values.

When sending data to a player, it is done immediately, but for every entity it increases the counter. If the counter hits the limit, you put the rest of remainder of the data in the queue.

Every tick, you reset all counters to 0, and you try to work away the queue from low to high index. If in the tick you hit the limit, you just break and leave the rest for the next tick.

Reply:
Im not sure if im getting what you're saying so sorry if i misunderstood. But i dont see how this could lead to a race condition, timers execute in the same sync thread as all the other code, so the timer loop will never be going on while `startNetWriteQueue` is being called and vice versa. As for the immediate sending of data that wouldn't fix the issue described here https://github.com/FPtje/Falcos-Prop-protection/pull/325#issuecomment-1975209906